helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -351,21 +347,16 @@ private void addToAssignmentRecord(AssignableReplica replica) {
   }
 
   private void updateCapacityAndUtilization(String capacityKey, int valueToSubtract) {
-    if (_currentCapacityMap.containsKey(capacityKey)) {
```

Comment:
Why remove this check? If the map does not contain the capacityKey, we will have NPE.

Note the replica might have different capacity keys. Depends on our implementation elsewhere, it might become possible to hit NPE.

Reply:
I don't think we need to check key existence at least in the private method since the class has total control of data input to the method. I could change and check the key exists in the caller method.

Reply:
The input is from AssignableReplica capacity keys. The AssignableNode class has no control.
You can also check there. But I think checking here is easier.

Reply:
+1 we should keep the check.

Reply:
Sure, added the check back.